adobe_vipm/adobe/errors.py

```
@@ -81,9 +83,68 @@ def __repr__(self) -> str:
+    Args:
+        error: The HTTP error raised for the error response.
+
+    Raises:
+        AdobeAPIError: When the response body is the documented Adobe error JSON.
+        AdobeHttpError: When the response body is not JSON.
+    """
+    logger.error(error)
+    try:  # noqa: WPS328, WPS505
+        raise AdobeAPIError(error.response.status_code, error.response.json())
+    except JSONDecodeError:
+        raise AdobeHttpError(error.response.status_code, error.response.content.decode())
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)